streams/src/main/java/org/apache/kafka/streams/kstream/internals/KTableImpl.java

```diff
@@ -220,6 +223,47 @@ public String queryableStoreName() {
         return doMapValues(mapper, new MaterializedInternal<>(materialized, builder, MAPVALUES_NAME));
     }
 
+    @Override
+    public <VR> KTable<K, VR> transformValues(ValueTransformerWithKeySupplier<? super K, ? super V, ? extends VR> transformerSupplier, String... stateStoreNames) {
+        return doTransformValues(transformerSupplier, null, stateStoreNames);
+    }
+
+    @Override
+    public <VR> KTable<K, VR> transformValues(final ValueTransformerWithKeySupplier<? super K, ? super V, ? extends VR> transformerSupplier,
+                                              final Materialized<K, VR, KeyValueStore<Bytes, byte[]>> materialized,
+                                              final String... stateStoreNames) {
+        Objects.requireNonNull(materialized, "materialized can't be null");
+        return doTransformValues(transformerSupplier, materialized, stateStoreNames);
+    }
+
+    private <VR> KTable<K, VR> doTransformValues(final ValueTransformerWithKeySupplier<? super K, ? super V, ? extends VR> transformerSupplier,
+                                               final Materialized<K, VR, KeyValueStore<Bytes, byte[]>> materialized,
+                                               final String... stateStoreNames) {
+        Objects.requireNonNull(stateStoreNames, "stateStoreNames");
+
+        final String name = builder.newProcessorName(TRANSFORMVALUES_NAME);
+
+        final MaterializedInternal<K, VR, KeyValueStore<Bytes, byte[]>> materializedInternal =
+                materialized == null ? null : new MaterializedInternal<>(materialized, builder, TRANSFORMVALUES_NAME);
+
+        final String queryableName = materializedInternal == null ? null : materializedInternal.storeName();
+
+        final KTableProcessorSupplier<K, V, VR> processorSupplier = new KTableTransformValues<>(this, transformerSupplier, queryableName);
+        builder.internalTopologyBuilder.addProcessor(name, processorSupplier, this.name);
+
+        if (stateStoreNames.length > 0) {
+            builder.internalTopologyBuilder.connectProcessorAndStateStores(name, stateStoreNames);
+        }
+
+        if (materializedInternal != null) {
+            builder.internalTopologyBuilder.addStateStore(
+                    new KeyValueStoreMaterializer<>(materializedInternal).materialize(),
+                    name);
+        }
+
+        return new KTableImpl<>(builder, name, processorSupplier, sourceNodes, queryableStoreName, isQueryable);
```

Comment:
If the resulted KTable is not materialized, we should 1) set the `queryableStoreName` to the current queryableStoreName; and 2) set is Queryable to `materialized != null && materialized.isQueryable()`. Please see `doFilter` and `doMapValues` for references.

Reply:
Good spot! I did reference these for methods, but they've recently been changed. I'll update accordingly.